macOS/DuckDuckGo/Preferences/View/PreferencesAboutView.swift

```diff
@@ -292,30 +296,38 @@ extension Preferences {
 
         @ViewBuilder
         private var updateButton: some View {
-            switch model.updateState {
-            case .upToDate:
-                Button(UserText.checkForUpdate) {
-                    model.checkForUpdate()
-                }
-                .buttonStyle(UpdateButtonStyle(enabled: true))
-            case .updateCycle(let progress):
-                if hasPendingUpdate {
-                    Button(model.areAutomaticUpdatesEnabled ? UserText.restartToUpdate : UserText.runUpdate) {
-                        model.runUpdate()
-                    }
-                    .buttonStyle(UpdateButtonStyle(enabled: true))
-                } else if progress.isFailed {
-                    Button(UserText.retryUpdate) {
+            if model.autoRestartAllowed {
+                switch model.updateState {
+                case .upToDate:
+                    Button(UserText.checkForUpdate) {
                         model.checkForUpdate()
                     }
                     .buttonStyle(UpdateButtonStyle(enabled: true))
-                } else {
-                    Button(UserText.checkForUpdate) {
-                        model.checkForUpdate()
+                case .updateCycle(let progress):
+                    if hasPendingUpdate {
+                        Button(model.areAutomaticUpdatesEnabled ? UserText.restartToUpdate : UserText.runUpdate) {
+                            model.runUpdate()
+                        }
+                        .buttonStyle(UpdateButtonStyle(enabled: true))
+                    } else if progress.isFailed {
+                        Button(UserText.retryUpdate) {
+                            model.checkForUpdate()
+                        }
+                        .buttonStyle(UpdateButtonStyle(enabled: true))
+                    } else {
+                        Button(UserText.checkForUpdate) {
+                            model.checkForUpdate()
+                        }
+                        .buttonStyle(UpdateButtonStyle(enabled: false))
+                        .disabled(true)
                     }
-                    .buttonStyle(UpdateButtonStyle(enabled: false))
-                    .disabled(true)
                 }
+            } else {
+                let configuration = model.updateButtonConfiguration
+
+                Button(configuration.title, action: configuration.action)
+                    .buttonStyle(UpdateButtonStyle(enabled: configuration.enabled))
+                    .disabled(!configuration.enabled)
```

Comment:
The diff is ugly, but this is what changes... if the feature flag is ON we now use this logic here.